### PR TITLE
Add the ability to switch on chars

### LINKED ISSUE #387: Switch statement with char cases throws compile exception
Trying to switch on a char value, like so:

    char c = ...;
    switch (c)
    {
        case '[':  // Line which errors
        {
            ...
            break;
        }
        ...
    }

Produces the following invalid cast exception during compilation:

    Error	4	INTERNAL ERROR: System.InvalidCastException: Invalid cast from 'Char' to 'Double'.
       at System.Char.System.IConvertible.ToDouble(IFormatProvider provider)
       at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
       at System.Convert.ChangeType(Object value, Type conversionType)
       at Saltarelle.Compiler.Compiler.StatementCompiler.VisitSwitchStatement(SwitchStatement switchStatement)
       at ICSharpCode.NRefactory.CSharp.SwitchStatement.AcceptVisitor(IAstVisitor visitor)
       at Saltarelle.Compiler.Compiler.StatementCompiler.VisitChildren(AstNode node)
       at Saltarelle.Compiler.Compiler.StatementCompiler.VisitBlockStatement(BlockStatement blockStatement)
       at ICSharpCode.NRefactory.CSharp.BlockStatement.AcceptVisitor(IAstVisitor visitor)
       at Saltarelle.Compiler.Compiler.StatementCompiler.Compile(Statement statement). Please report this as an issue on https://github.com/erik-kallen/SaltarelleCompiler/

### PR DESCRIPTION
**Pull request by jon-hall**
Fixes #387.

### CODE REVIEW

File: Compiler/Saltarelle.Compiler.Tests/CompilerTests/MethodCompilation/Statements/SwitchTests.cs [hunk 1]
```diff
@@ -92,6 +92,29 @@ public class SwitchTests : MethodCompilerTestBase {
...
+@"	switch ($s) {
+		case 'X': {
+			var $x = 0;
+			break;
+		}
+	}
+");
+        }
+
 		[Test]
```
Comment (erik-kallen) [inline Compiler/Saltarelle.Compiler.Tests/CompilerTests/MethodCompilation/Statements/SwitchTests.cs]: chars are treated as numeric values, so the proper translation is ```case 88:``` (unicode value of 'X').

File: Compiler/Saltarelle.Compiler/Compiler/StatementCompiler.cs [hunk 2]
```diff
@@ -1046,7 +1046,11 @@ private class GatherGotoCaseAndDefaultDataVisitor : DepthFirstAstVisitor {
 						}
 						else if (value is string) {
 							values.Add(JsExpression.String((string)value));
-						}
+                        }
+                        else if (value is char)
+                        {
+                            values.Add(JsExpression.String(value.ToString()));
+                        }
```
Comment (erik-kallen) [inline Compiler/Saltarelle.Compiler/Compiler/StatementCompiler.cs]: Can you please use tabs instead of spaces?

**Comment by erik-kallen** [comment]
Thanks. Some comments (inline), though

**Comment by jon-hall** [comment]
Apologies for the mistake with translating chars to strings instead of numerics in the js, has been corrected now, along with the tabbing.  Cheers.

**Comment by erik-kallen** [comment]
Thanks!

**Pull request merged**